### PR TITLE
Issue #11710: Disable IDEA Inspection for AnnotationClass

### LINKED ISSUE #11710: Disable IDEA Inspection for  `AnnotationClass`
The `AnnotationClass` inspection is a "Java Language Level" issue, which pertains to Java versions less than 1.5 (Java 5). Since we only support jdk/jre >= 11, this inspection is not relevant to the project. Sadly, the IDEA inspections website has no information about this inspection, but from the "Inspections" view in Intellij: 

`Reports annotation interfaces. Such interfaces are not supported in Java 1.4 and earlier JVM. `

~~The `'assert' statement` inspection pertains to Java versions less than 1.3 (Java 3). Since we only support jdk/jre >= 11, this inspection is not relevant to the project. Sadly, the IDEA inspections website also has no information about this inspection, but from the "Inspections" view in Intellij:~~

~~`Reports assert statements. By default, assert statements are disabled during execution in the production environment. Consider using logger or exceptions instead.
The assert statements are not supported in Java 1.3 and earlier JVM. `~~

We can safely remove ~~these two~~ this inspection since we do not support language levels below 11.

### PR DESCRIPTION
**Pull request by nick-mancuso**
Closes #11710

### CODE REVIEW

File: config/intellij-idea-inspections.xml [hunk 1]
```diff
@@ -629,8 +629,8 @@
                    enabled_by_default="false"/>
   <!-- this rule is inspection is only for old java, we are ok to use annotations -->
   <inspection_tool class="Annotation" enabled="false" level="ERROR" enabled_by_default="false"/>
-  <inspection_tool class="AnnotationClass" enabled="true" level="ERROR"
-                   enabled_by_default="true"/>
+  <inspection_tool class="AnnotationClass" enabled="false" level="ERROR"
+                   enabled_by_default="false"/>
```
Comment (romani) [inline config/intellij-idea-inspections.xml:633]: Please put comment on reason of disablement

**Review by romani — CHANGES_REQUESTED** [review]
Items

File: config/intellij-idea-inspections.xml (hunk 1 above)
Comment (nick-mancuso) [reply config/intellij-idea-inspections.xml:634]: Done

**Review by romani — APPROVED** [review]
Ok to merge

File: src/main/java/com/puppycrawl/tools/checkstyle/XdocsPropertyType.java [hunk 2]
```diff
@@ -29,7 +29,6 @@
  * for clarification of the user-friendly type.
  * For example, some string array property may represent a set of tokens.
  *
```
Comment (rnveach) [inline src/main/java/com/puppycrawl/tools/checkstyle/XdocsPropertyType.java:31]: Please remove empty ending line for javadoc.

**Review by rnveach — CHANGES_REQUESTED** [review]

File: src/main/java/com/puppycrawl/tools/checkstyle/XdocsPropertyType.java (hunk 2 above)
Comment (nick-mancuso) [reply src/main/java/com/puppycrawl/tools/checkstyle/XdocsPropertyType.java]: Done

**Review by rnveach — APPROVED** [review]

File: config/intellij-idea-inspections.xml (hunk 1 above)
Comment (pbludov) [reply config/intellij-idea-inspections.xml:634]: "this rule is inspection is only for old java" -> "this inspection is only for old java"?

**Comment by romani** [comment]
rebased.

**Pull request merged**